buildpack_store.go

```diff
@@ -86,12 +96,20 @@ type BuildpackStoreGet struct {
 	cacheManager CacheManager
 	local        LocalFetcher
 	remote       RemoteFetcher
+	oci          BPStoreDockerPull
 
 	offline bool
 	version string
 }
 
 func (g BuildpackStoreGet) Execute(url string) (string, error) {
+	if strings.HasPrefix(url, "docker://") {
+		if g.oci == nil {
+			return "", fmt.Errorf("must provide OCI fetcher to fetch OCI images")
```

Comment:
This is kind of gross, but it's the only way I could see to make this backwards-compatible and avoid a breaking change in the exported function signatures.

The OCI fetcher has to be injected, so either we force the user to inject it at construction time (i.e. `NewBuildpackStore(dockerPull BPStoreDockerPull)`) which is a breaking change, or we accept that it might be `nil` when we try to use it.